kernel/src/hil/adc.rs

```diff
@@ -14,19 +14,32 @@ pub trait Adc {
 
     /// Request a single ADC sample on a particular channel.
     /// Used for individual samples that have no timing requirements.
+    /// All ADC samples will be the raw ADC value left-justified in the u16.
     fn sample(&self, channel: &Self::Channel) -> ReturnCode;
 
     /// Request repeated ADC samples on a particular channel.
     /// Callbacks will occur at the given frequency with low jitter and can be
     /// set to any frequency supported by the chip implementation. However
     /// callbacks may be limited based on how quickly the system can service
     /// individual samples, leading to missed samples at high frequencies.
+    /// All ADC samples will be the raw ADC value left-justified in the u16.
     fn sample_continuous(&self, channel: &Self::Channel, frequency: u32) -> ReturnCode;
 
     /// Stop a sampling operation.
     /// Can be used to stop any simple or high-speed sampling operation. No
     /// further callbacks will occur.
     fn stop_sampling(&self) -> ReturnCode;
+
+    /// Function to ask the ADC how many bits of resolution are in the samples
+    /// it is returning.
+    fn get_resolution_bits(&self) -> usize;
+
+    /// Function to ask the ADC what reference voltage it used when taking the
+    /// samples. This allows the user of this interface to calculate an actual
+    /// voltage from the ADC reading.
+    ///
+    /// The returned reference voltage is in millivolts.
```

Comment:
is in millivolts, or 0 if unknown.

Reply:
Would it be more "rust"-y to return an `Option<usize>` and `None` if unknown?

Reply:
Either of these seems fine to me. `get_resolution_bits` should work the same way though.

Reply:
Hmm, why should the reference voltage be optional then? If the suggestion is that the reference could be an external pin that the ADC doesn't know, that could be passed into the driver at init time by the board `main.rs` configuration.

Reply:
- That value could instead be passed to whatever would be using `get_reference_voltage()`.
- If the HW doesn't know, then conceptually it makes sense that it should return that it doesn't know.
- The reference voltage could change at runtime.